src/store.cc

```diff
@@ -699,7 +707,7 @@ StoreEntry::adjustVary()
         // certain conditions. If those conditions do not apply to Vary markers,
         // then refactor to call storeCreatePureEntry() above.  Otherwise,
         // refactor to simply check whether `pe` is already public below.
-        if (!pe->makePublic()) {
+        if (!pe->makePublic(false)) {
```

Comment:
We have discussed that added `makePublic(false)` deserve analysis. I think this particular case matches the suggested "share if possible" default -- there is no point in creating a Vary marker object if that object is not going to be shared beyond the current transaction, right?